### ISSUE TITLE
Can you share the python code to extract the  argument counts and types?

### ISSUE THREAD

**Issue by qingbol** [shensq04/EKLAVYA#5]
Can you share the python code to extract the  argument counts and types? I appreciate it if you can.

**Comment by bbaker1229** [comment]
Yes.  I would also be interested in the python code used to create the pickle files.

**Comment by melynx** [comment]
I'm sorry, what do you mean the argument counts and types? 
You can just obtain the tensor from the model and evaluate it.

**Comment by bbaker1229** [comment]
In GitHub you have pickles and clean pickles which contain the functions,
architecture compiled for etc. I would like to know how those files were
created.

Thank you for your time,
Bryan Baker.

**Comment by tforhan** [comment]
I don't know for sure if this is what the original commenter wanted, but I and some colleagues were hoping to get the code that was used to parse binaries and create the pickles for them. We would like to be able to compile our own binaries with other compilers (but still include the DWARF information) and then create pickles for them that can be used to train/evaluate the neural network.

**Comment by FantaXuan** [comment]
Same as tforhan, I'm curious about the code that can get argument counts and types from binary using pyelftools. I guess there are so many complex and corner cases even only for C program.

**Comment by bbaker1229** [comment]
I was also stuck on this for a project that I did for school a few years ago.  I ended up writing this program after reverse engineering the pickle file structure used for this neural network.  I have not used it since so I can not speak for the function of it but maybe it will help you.  

https://gist.githubusercontent.com/bbaker1229/a0002fa30c1cbee908faac0791adfa7c/raw/2849c65e7d8ab8a0329fa60bc4cf11276c2a76a4/gccbinary2pickle.py
